projects/batfish-client/test/org/batfish/client/ClientTest.java

```diff
@@ -0,0 +1,288 @@
+   }
+
+   @Test
+   public void testUnStringInputWhenExpectString() throws IOException {
+      JsonNode numberNode = mapper.readTree("10");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.STRING);
+      thrown.expect(BatfishException.class);
+      errorMessage = generateErrorMessage("10", numberNode.getNodeType(),
+            variable.getType());
+      thrown.expectMessage(errorMessage);
+      Client.validateType(numberNode, variable);
+   }
+
+   @Test
+   public void testInvalidBooleanValue() throws IOException {
+      JsonNode invalidBooleanNode = mapper.readTree("\"true\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.BOOLEAN);
+      thrown.expect(BatfishException.class);
+      errorMessage = generateErrorMessage("\"true\"",
+            invalidBooleanNode.getNodeType(), variable.getType());
+      thrown.expectMessage(errorMessage);
+      Client.validateType(invalidBooleanNode, variable);
+   }
+
+   @Test
+   public void testValidBooleanValue() throws IOException {
+      JsonNode booleanNode = mapper.readTree("true");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.BOOLEAN);
+      try {
+         Client.validateType(booleanNode, variable);
+      } catch (Exception e) {
+         throw new BatfishException("Unexpected exception");
+      }
+   }
+
+   @Test
+   public void testInvalidComparatorValue() throws IOException {
+      JsonNode invalidComparator = mapper.readTree("\"=>\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.COMPARATOR);
+      thrown.expect(BatfishException.class);
+      errorMessage = generateErrorMessage("\"=>\"",
+            invalidComparator.getNodeType(), variable.getType());
+      thrown.expectMessage(errorMessage);
+      Client.validateType(invalidComparator, variable);
+   }
+
+   @Test
+   public void testValidComparatorValue() throws IOException {
+      JsonNode comparatorNode = mapper.readTree("\">=\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.COMPARATOR);
+      try {
+         Client.validateType(comparatorNode, variable);
+      } catch (Exception e) {
+         throw new BatfishException("Unexpected exception");
+      }
+   }
+
+   @Test
+   public void testInvalidIntegerValue() throws IOException {
+      JsonNode invalidInteger = mapper.readTree("1.5");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.INTEGER);
```

Comment:
there seems to be a lot of code duplication in this file .. the first 4 lines of every test are almost the same. Can you think of a way that utility functions could make this cleaner?